### PR TITLE
[notebook2] home page

### PR DESCRIPTION
**Pull request by akotlar**
Stacks on #5448. Just moves `/` to `/notebook` (` index.html` to `notebook.html`), and adds a home page with the bouncy Hail that people liked. Spoke with Cotton, he wanted it.

Only other change is adding cursor:pointer on `input[type="submit"]` so that the "Start a Hail Notebook" button shows a hand. 

edit2: Added 2 more commits, +7 -6 lines total, to fix login redirect link, and remove the /user link (which didn't work), to make this completely working as of this PR.
430a01902c7994d64c4b2986f1b174fb3f32bbb9
6a4c842c6a8e7a23a83cb7f624974a61e6f85962

cc @cseed

### CODE REVIEW

File: notebook2/notebook/templates/notebook.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <form action="{{ form_action_url }}" method="post">
+          <p>
+            <label for="image">Image:</label>
+            <select name="image" id="image">
+              {% for image in images %}
+              <option value="{{ image }}"{{'selected' if image == default else ''}}>{{ image }}</option>
+              {% endfor %}
+            </select>
+          </p>
+          <input type="submit" value="Start a Hail Notebook" />
```
Comment (danking) [inline notebook2/notebook/templates/notebook.html]: I think we should just use a `button` tag. AFAICT, a `button` tag (under the description of the type attribute):
> [...] submits the form data to the server. This is the default [...]

[MDN Button Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button)

Comment (akotlar) [reply notebook2/notebook/templates/notebook.html]: input type=html is the canonical way afaik for html forms: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/submit 

"
input type="submit" buttons are used to submit forms. If you want to create a custom button and then customize the behavior using JavaScript, you need to use input type="button", or better still, a button element.
"

**Comment by akotlar** [comment]
> Can you share the long term vision for this page? Will notebook become a generic `app.hail.is` with many tabs? Changing the index page to a page with no content seems strictly less helpful to the user. They must make an extra click after they type `notebook.hail.is` into their URL bar to get to a login form. If I was the user, I'd want to get dumped right into the log-in form (but maybe I'm a weird user?).

Notebook will host the batch stuff. I have no comments about vision longer term than that, it's undefined as far as I know (meaning, I don't know)

We can get rid of the home page as well. I ran the home page idea by @cseed but maybe I misunderstood.

File: notebook2/notebook/templates/notebook.html (hunk 1 above)
Comment (akotlar) [reply notebook2/notebook/templates/notebook.html]: Additional comment on this: use of `input type='submit'` predates `button`. I remember using the `button` for styling reasons...it really doesn't matter to me which one we use.

Notebook 1 used type='submit', so I kept it. If you'd like me to change that in this PR, no problem!

**Pull request merged**